testutils/lint: TestNightlyLint failed

testutils/lint.TestNightlyLint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_LintURLsBazel/20407929?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_LintURLsBazel/20407929?buildTab=artifacts#/) on release-24.1.23-rc @ [b2d9e467bbe18f26450bf5600c249d2b59765b9a](https://github.com/cockroachdb/cockroach/commits/b2d9e467bbe18f26450bf5600c249d2b59765b9a):


```
I250829 05:12:59.913829 29 (gostd) urlcheck.go:230  [-] 11  Checking https://www.cockroachlabs.com/docs/v24.1/table-expressions.html...
I250829 05:12:59.913893 32 (gostd) urlcheck.go:230  [-] 12  Checking https://www.cockroachlabs.com/docs/v24.1/show-partitions.html...
I250829 05:12:59.913955 27 (gostd) urlcheck.go:230  [-] 13  Checking https://www.cockroachlabs.com/docs/v24.1/insert.html...
I250829 05:12:59.914029 33 (gostd) urlcheck.go:230  [-] 14  Checking https://www.cockroachlabs.com/docs/v24.1/revoke.html...
I250829 05:12:59.914080 24 (gostd) urlcheck.go:230  [-] 15  Checking https://www.cockroachlabs.com/docs/v24.1/alter-index.html...
I250829 05:12:59.914171 66 (gostd) urlcheck.go:230  [-] 16  Checking https://www.cockroachlabs.com/docs/v24.1/show-backup.html...
I250829 05:12:59.914218 25 (gostd) urlcheck.go:230  [-] 17  Checking https://www.cockroachlabs.com/docs/v24.1/import-into.html...
I250829 05:12:59.914282 28 (gostd) urlcheck.go:230  [-] 18  Checking https://www.cockroachlabs.com/docs/v24.1/create-view.html...
I250829 05:12:59.914320 71 (gostd) urlcheck.go:230  [-] 19  Checking https://www.cockroachlabs.com/docs/v24.1/set-vars.html...
I250829 05:12:59.914387 26 (gostd) urlcheck.go:230  [-] 20  Checking https://www.cockroachlabs.com/docs/v24.1/create-function.html...
I250829 05:12:59.955240 233 (gostd) urlcheck.go:230  [-] 21  Checking https://www.cockroachlabs.com/docs/v24.1/show-tables.html...
I250829 05:13:00.177592 190 (gostd) urlcheck.go:230  [-] 22  Checking https://www.cockroachlabs.com/docs/v24.1/create-index.html...
I250829 05:13:00.219985 240 (gostd) urlcheck.go:230  [-] 23  Checking https://www.cockroachlabs.com/docs/v24.1/create-procedure.html...
I250829 05:13:00.224796 191 (gostd) urlcheck.go:230  [-] 24  Checking https://www.cockroachlabs.com/docs/v24.1/show-users.html...
I250829 05:13:00.225088 241 (gostd) urlcheck.go:230  [-] 25  Checking https://www.cockroachlabs.com/docs/v24.1/alter-database.html...
I250829 05:13:00.241997 192 (gostd) urlcheck.go:230  [-] 26  Checking https://www.cockroachlabs.com/docs/v24.1/drop-table.html...
I250829 05:13:00.262621 258 (gostd) urlcheck.go:230  [-] 27  Checking https://www.cockroachlabs.com/docs/v24.1/delete.html...
I250829 05:13:00.272701 193 (gostd) urlcheck.go:230  [-] 28  Checking https://www.cockroachlabs.com/docs/v24.1/select-clause.html...
I250829 05:13:00.279799 259 (gostd) urlcheck.go:230  [-] 29  Checking https://www.cockroachlabs.com/docs/v24.1/rollback-transaction.html...
I250829 05:13:00.283514 274 (gostd) urlcheck.go:230  [-] 30  Checking https://www.cockroachlabs.com/docs/v24.1/alter-table.html...
I250829 05:13:00.285561 260 (gostd) urlcheck.go:230  [-] 31  Checking https://www.cockroachlabs.com/docs/v24.1/show-create.html...
I250829 05:13:00.289111 275 (gostd) urlcheck.go:230  [-] 32  Checking https://www.cockroachlabs.com/docs/v24.1/alter-partition.html...
I250829 05:13:00.321433 261 (gostd) urlcheck.go:230  [-] 33  Checking https://www.cockroachlabs.com/docs/v24.1/alter-procedure.html...
I250829 05:13:00.322526 276 (gostd) urlcheck.go:230  [-] 34  Checking https://www.cockroachlabs.com/docs/v24.1/drop-procedure.html...
I250829 05:13:00.324264 262 (gostd) urlcheck.go:230  [-] 35  Checking https://www.cockroachlabs.com/docs/v24.1/show-grants.html...
I250829 05:13:00.330211 277 (gostd) urlcheck.go:230  [-] 36  Checking https://www.cockroachlabs.com/docs/v24.1/backup.html...
I250829 05:13:00.331434 263 (gostd) urlcheck.go:230  [-] 37  Checking https://www.cockroachlabs.com/docs/v24.1/commit-transaction.html...
I250829 05:13:00.344269 278 (gostd) urlcheck.go:230  [-] 38  Checking https://www.cockroachlabs.com/docs/v24.1/drop-database.html...
I250829 05:13:00.354184 264 (gostd) urlcheck.go:230  [-] 39  Checking https://www.cockroachlabs.com/docs/v24.1/drop-function.html...
I250829 05:13:00.354277 279 (gostd) urlcheck.go:230  [-] 40  Checking https://www.cockroachlabs.com/docs/v24.1/grant.html...
I250829 05:13:00.361472 158 (gostd) urlcheck.go:230  [-] 41  Checking https://www.cockroachlabs.com/docs/v24.1/set-transaction.html...
I250829 05:13:00.371389 280 (gostd) urlcheck.go:230  [-] 42  Checking https://www.cockroachlabs.com/docs/v24.1/restore.html...
I250829 05:13:00.376036 159 (gostd) urlcheck.go:230  [-] 43  Checking https://www.cockroachlabs.com/docs/v24.1/cluster-settings.html...
I250829 05:13:00.452311 281 (gostd) urlcheck.go:230  [-] 44  Checking https://www.cockroachlabs.com/docs/v24.1/update.html...
I250829 05:13:00.543306 160 (gostd) urlcheck.go:230  [-] 45  Checking https://www.cockroachlabs.com/docs/v24.1/show-index.html...
I250829 05:13:00.557929 282 (gostd) urlcheck.go:230  [-] 46  Checking https://www.cockroachlabs.com/docs/v24.1/begin-transaction.html...
I250829 05:13:00.558075 161 (gostd) urlcheck.go:230  [-] 47  Checking https://www.cockroachlabs.com/docs/v24.1/alter-function.html...
I250829 05:13:00.566929 290 (gostd) urlcheck.go:230  [-] 48  Checking https://www.cockroachlabs.com/docs/v24.1/savepoint.html...
I250829 05:13:00.567459 265 (gostd) urlcheck.go:230  [-] 49  Checking https://www.cockroachlabs.com/docs/v24.1/show-constraints.html...
I250829 05:13:00.582988 291 (gostd) urlcheck.go:230  [-] 50  Checking https://www.cockroachlabs.com/docs/v24.1/create-database.html...
I250829 05:13:00.584899 266 (gostd) urlcheck.go:230  [-] 51  Checking https://www.cockroachlabs.com/docs/v24.1/show-vars.html...
I250829 05:13:00.587514 292 (gostd) urlcheck.go:230  [-] 52  Checking https://www.cockroachlabs.com/docs/v24.1/show-default-privileges...
I250829 05:13:00.606778 267 (gostd) urlcheck.go:230  [-] 53  Checking https://www.cockroachlabs.com/docs/v24.1/create-table-as.html...
I250829 05:13:00.608814 293 (gostd) urlcheck.go:230  [-] 54  Checking https://www.cockroachlabs.com/docs/v24.1/show-zone-configurations.html...
I250829 05:13:00.610580 268 (gostd) urlcheck.go:230  [-] 55  Checking https://www.cockroachlabs.com/docs/v24.1/upsert.html...
    nightly_lint_test.go:41: https://en.wikipedia.org/wiki/Cron : 403 Forbidden
             611:  See https://en.wikipedia.org/wiki/Cron
             1544:  See https://en.wikipedia.org/wiki/Cron
        1 errors
    --- FAIL: TestNightlyLint/TestHelpURLs (1.07s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestNightlyLint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53992

--- a/pkg/cmd/urlcheck/lib/urlcheck/urlcheck.go
+++ b/pkg/cmd/urlcheck/lib/urlcheck/urlcheck.go
@@ -104,7 +104,7 @@ func chompUnbalanced(left, right rune, s string) string {
 }
 
 func checkURL(client *http.Client, url string) error {
-	resp, err := client.Head(url)
+	resp, err := httpDo(client, "HEAD", url)
 	if err != nil {
 		return err
 	}
@@ -121,7 +121,7 @@ func checkURL(client *http.Client, url string) error {
 	// for any other error. Still, we link to several misconfigured servers that
 	// return 403 Forbidden or 500 Internal Server Error for HEAD requests, but
 	// not for GET requests.
-	resp, err = client.Get(url)
+	resp, err = httpDo(client, "GET", url)
 	if err != nil {
 		return err
 	}
@@ -134,6 +134,21 @@ func checkURL(client *http.Client, url string) error {
 	}
 
 	return errors.Newf("%s", errors.Safe(resp.Status))
+}
+
+// N.B. we set custom User-Agent header to avoid being blocked.
+// E.g., as of 08/25/25, Wikipedia blocks default UAs [1].
+// [1] https://phabricator.wikimedia.org/T400119
+func httpDo(c *http.Client, requestType string, url string) (resp *http.Response, err error) {
+	req, err := http.NewRequest(requestType, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	// This UA seems to comply with Wikipedia's policy [1].
+	// [1] https://foundation.wikimedia.org/wiki/Policy:Wikimedia_Foundation_User-Agent_Policy
+	req.Header.Set("User-Agent", "MyGoApplication/1.0 (https://example.com/myapp; myapp@example.com)")
+
+	return c.Do(req)
 }
 
 func checkURLWithRetries(client *http.Client, url string) error {

